sympy/complexanalysis/Contour Integration.py

```diff
@@ -0,0 +1,39 @@
+from sympy import *
+import matplotlib.pyplot as plt
+
+class ComplexIntegral:
+
+    def __init__(self, eq, params=dict(shape="semi-circle", origin=(0,0), radius=oo, angle=pi)):
+
+        self.eq = eq
+        self.shape = params["shape"]
+        if "circle" in self.shape:
+            self.solveit = self.solveit_cir()
+            self.origin = params["origin"]
+            self.rad = params["radius"]
+            self.angle = params["angle"]
+
+        else:
+            pass 
+            #Enter in anticlockwise direction
+            self.solveit = solveit_oth()
+            self.points = points
+
+    def contour(self):
+
+        #Plot the contour
+        pass   
+
+    def residue_calc(self, rooot, mult, x ):
```

Comment:
There is already a `residue` function that can be used here.

Reply:
Ok, I will substitute it here. But should I work on it?

Reply:
Can it be a good enough project for GSoC?